core/core-impl/src/main/java/ai/timefold/solver/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -1039,6 +1039,7 @@ public long getMaximumValueCount(Solution_ solution) {
     /**
      * Calculates an indication on how big this problem instance is.
      * This is intentionally very loosely defined for now.
+     * This is explicitly NOT the solution space size.
```

Comment:
I see that this PR makes no changes to any of the scale computations. So we are happy with what is currently there for list and chained? I recall there were some PLANNER JIRA references there; at the very least they should be removed, since we've been over that code and the new version is something we want to move forward with.

Reply:
Was waiting on @ge0ffrey opinion since changing this code would result in problem scale changes in benchmark. I have changes locally that make this calculation more well defined:

- log is in base maximumValueRangeSize (for basic variables, most entities will either add 0 (pinned) or 1 (not pinned) to the count. If the value range for the entity is entity sourced, it will add a number between 0 and 1.
- For list/chained, the formula is `(V! * (V * E))/E!`, where `V` the value count and `E` the entity count (There are `V!` ways to permute a list of `V` elements. You can place `E` markers on a list of size `V` `V * E` ways. You can arrange `E` partitions of a list `E!` ways. So there are `V! * (V * E)` ways to create `E` partitions for all permutations of a list of size `V`, of which   `V! * (V * E)/E!` are unique.

The base of the log in the record is dynamic (i.e. based on `maximumValueRangeSize`) to preserve accuracy, but can be converted to a common base using the formula `log_b(x) = log_a(x) / log_a(b)`